graph-service/analyze.py

```
@@ -0,0 +1,115 @@
+                """,
+            }
+            # Drop old projection if exists (handle bool, DataFrame, or Series return types without ambiguous truth-value errors)
+            exists_resp = gds.graph.exists(GDS_GRAPH_NAME)
+            graph_exists = False
+            if isinstance(exists_resp, bool):
+                graph_exists = exists_resp
+            else:
+                # For DataFrame/Series-like results, try to access an "exists" field safely
+                try:
+                    if hasattr(exists_resp, "empty") and not exists_resp.empty:
+                        if "exists" in exists_resp:
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)